### PR TITLE
fix(build): switch to esbuild and tsc instead of pika

### PR DESCRIPTION
**Pull request by kfcampbell**
Replicating [octokit/request.js#584](https://github.com/octokit/request.js/pull/584) to another repo. 

---

Pika has been deprecated for a while now, and the project has now been archived, and because of that the released npm package is missing most of the files generated by the build step.

Uses esbuild to transpile the TS source code into an ESM source, NodeJs bundle, and a browser bundle

Uses the TypeScript compiler to generate the types.

### Before the change?

* Uses pika for the build system
* The released npm package is missing most of the files generated by the build step. `dist-node`, `dist-types`, `dist-web`... even though they are generated correctly.

### After the change?

* Uses esbuild to generate the JS bundles
* Uses the TypeScript compiler to generate the type definitions
* The correct file matching patterns are used so we publish all the necessary files again.

----

### Pull request checklist
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Added the appropriate label for the given change

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/main/community/breaking_changes.md) to help!

- [ ] Yes (Please add the `Type: Breaking change` label)
- [x] No

If `Yes`, what's the impact:

* N/A

### Pull request type

Please add the corresponding label for change this PR introduces:
- Bugfix: `Type: Bug`
- Dependencies/code cleanup: `Type: Maintenance`

----

### CODE REVIEW

**Review by nickfloyd — DISMISSED** [review]
❤️

**Comment by wolfy1339** [comment]
Make sure to explicitly mark every type import and export.

**Comment by wolfy1339** [comment]
I don't think it's the best idea to move `@octokit/types` to `devDependencies`.

There was a reasoning of putting it in regular `dependencies`

If it's not in regular dependencies, then people using this package may not have types in their IDE

**Comment by kfcampbell** [comment]
Explicit imports/exports and the previously discussed `--verbatimModuleSyntax` flag have been added. I'm not quite sure what's going on with the Node 14 errors, though I can reproduce them locally.

File: src/endpoints-to-methods.ts [hunk 1]
```diff
@@ -1,13 +1,13 @@
 import { Octokit } from "@octokit/core";
```
Comment (wolfy1339) [inline src/endpoints-to-methods.ts:1]:
```suggestion
import type { Octokit } from "@octokit/core";
```

File: src/index.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import { Octokit } from "@octokit/core";
```
Comment (wolfy1339) [inline src/index.ts:1]:
```suggestion
import type { Octokit } from "@octokit/core";
```

**Review by wolfy1339 — COMMENTED** [review]
Octokit is only being used as a type here

**Comment by wolfy1339** [comment]
I think I see the problem, the problem is the lockfile here. It's using version 3 which isn't backwards compatible with older versions of npm

Can you regenerate it with Node v14?

**Comment by kfcampbell** [comment]
Thanks for the tip! Done.

**Review by wolfy1339 — APPROVED** [review]

**Comment by kfcampbell** [comment]
@wolfy1339 I've updated this PR with the latest version of octokit/tsconfig. Please take another look when you get a moment!

File: package.json [hunk 3]
```diff
@@ -56,6 +56,14 @@
   },
   "jest": {
     "preset": "ts-jest",
```
Comment (wolfy1339) [inline package.json:58]:
```suggestion
```

**Review by wolfy1339 — APPROVED** [review]

**Pull request merged**